batch/sql/add-user-resources-ready-cores-token.sql

```diff
@@ -0,0 +1,171 @@
+ALTER TABLE user_resources ADD COLUMN token INT NOT NULL DEFAULT 0;
+ALTER TABLE user_resources DROP PRIMARY KEY, ADD PRIMARY KEY(`user`, `token`);
+
+ALTER TABLE ready_cores ADD COLUMN token INT NOT NULL DEFAULT 0;
+ALTER TABLE ready_cores ADD PRIMARY KEY(`token`);
+
+CREATE TABLE IF NOT EXISTS `batches_staging` (
+  `batch_id` BIGINT NOT NULL,
+  `token` INT NOT NULL,
+  `n_jobs` INT NOT NULL DEFAULT 0,
+  `n_ready_jobs` INT NOT NULL DEFAULT 0,
+  `ready_cores_mcpu` BIGINT NOT NULL DEFAULT 0,
+  PRIMARY KEY (`batch_id`, `token`),
+  FOREIGN KEY (`batch_id`) REFERENCES batches(`id`) ON DELETE CASCADE
+) ENGINE = InnoDB;
```

Comment:
I think you should write queries to completely recompute ready_cores, user_resources and batch_staging as part of the migration.  I posted logic to recompute user_resources from a query, you can start with that.